Symfony/CS/Tests/Fixer/Contrib/Php4ConstructorFixerTest.php

```diff
@@ -0,0 +1,912 @@
+        var_dump(3);
+    }
+}
+EOF;
+
+        $this->makeTest($expected);
+    }
+
+    public function testPhp4Only()
+    {
+        $expected = <<<'EOF'
+<?php
+
+class Foo
+{
+    /**
+     * Constructor
+     */
+    function __construct($bar)
+    {
+        var_dump(1);
+    }
+
+    function bar()
+    {
+        var_dump(3);
+    }
+}
+EOF;
+
+        $input = <<<'EOF'
+<?php
+
+class Foo
+{
+    /**
+     * Constructor
+     */
+    function foO($bar)
+    {
+        var_dump(1);
+    }
+
+    function bar()
+    {
+        var_dump(3);
+    }
+}
+EOF;
+
+        $this->makeTest($expected, $input);
+    }
+
+    public function testBothTheRightWay1()
+    {
+        $expected = <<<'EOF'
+<?php
+
+class Foo
+{
+    /**
+     * Constructor
+     */
+    public function __construct()
+    {
+        var_dump(1);
+    }
+
+    public function bar()
+    {
+        var_dump(3);
+    }
+}
+EOF;
+
+        $input = <<<'EOF'
+<?php
+
+class Foo
+{
+    /**
+     * Constructor
+     */
+    public function __construct()
+    {
+        var_dump(1);
+    }
+
+    /**
+     * PHP-4 Constructor
+     */
+    function Foo()
+    {
+        // Call PHP5!
+        $this->__construct();
+    }
+
+    public function bar()
+    {
+        var_dump(3);
+    }
+}
+EOF;
+
+        $this->makeTest($expected, $input);
+    }
+
+    public function testBothTheRightWay2()
+    {
+        $expected = <<<'EOF'
+<?php
+
+class Foo
+{
+    /**
+     * Constructor
+     */
+    public function __construct($bar)
+    {
+        var_dump(1);
+    }
+
+    public function bar()
+    {
+        var_dump(3);
+    }
+}
+EOF;
+
+        $input = <<<'EOF'
+<?php
+
+class Foo
+{
+    /**
+     * Constructor
+     */
+    public function __construct($bar)
+    {
+        var_dump(1);
+    }
+
+    /**
+     * PHP-4 Constructor
+     */
+    function Foo($bar)
+    {
+        // Call PHP5!
+        $this->__construct($bar);
+    }
+
+    public function bar()
+    {
+        var_dump(3);
+    }
+}
+EOF;
+
+        $this->makeTest($expected, $input);
+    }
+
+    public function testBothTheRightWay3()
+    {
+        $expected = <<<'EOF'
+<?php
+
+class Foo
+{
+    /**
+     * Constructor
+     */
+    public function __construct($bar = 1, $baz = null)
+    {
+        var_dump(1);
+    }
+
+    public function bar()
+    {
+        var_dump(3);
+    }
+}
+EOF;
+
+        $input = <<<'EOF'
+<?php
+
+class Foo
+{
+    /**
+     * Constructor
+     */
+    public function __construct($bar = 1, $baz = null)
+    {
+        var_dump(1);
+    }
+
+    /**
+     * PHP-4 Constructor
+     */
+    function Foo($bar = 1, $baz = null)
+    {
+        // Call PHP5!
+        $this->__construct($bar, $baz);
+    }
+
+    public function bar()
+    {
+        var_dump(3);
+    }
+}
+EOF;
+
+        $this->makeTest($expected, $input);
+    }
+
+    public function testBothTheOtherWayAround()
+    {
+        $expected = <<<'EOF'
+<?php
+
+class Foo
+{
+
+    /**
+     * PHP-4 Constructor
```

Comment:
My reasonong was the following: if the "real" constructor is the PHP4 one, then it's the one that most likely contains any meaningful info in the docblock.